components/relevancy/src/rs.rs

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+use serde::Deserialize;
+
+/// The Remote Settings collection name.
+pub(crate) const REMOTE_SETTINGS_COLLECTION: &str = "content-relevance";
+
+/// A record in the Relevancy Remote Settings collection.
+#[derive(Clone, Debug, Deserialize)]
+pub struct RelevancyRecord {
+    #[serde(rename = "type")]
+    pub record_type: String,
+    pub record_custom_details: RecordCustomDetails,
+}
+
+// Custom details related to category of the record.
+#[derive(Clone, Debug, Deserialize)]
+pub struct RecordCustomDetails {
+    pub category_to_domains: CategoryToDomains,
```

Comment:
This isn't a comment on the code, and I don't think we need to address this now, but this extra level of nesting seems unnecessary to me.  Couldn't we just store the `CategoryToDomains` fields in `RecordCustomDetails`?  IRC `CategoryToDomains` used to be in a list, but now that it's a single child object I don't see the point.

Reply:
Yeah i agree this is clunky with the extra level of nesting.
This is what we have i our schema doc
```
// Example
{
  "id": "shopping-0001",
  "record_type": "category_to_domains",
  "record_custom_details": {
    "category_to_domains": {
      "category": "shopping",
      "category_code": 1,
      "version": 1
    }
  }
}

```

Reply:
To provide more context, we're reusing what we've done for Merino, see the ADR [here](https://github.com/mozilla-services/merino-py/blob/main/docs/adr/0002-merino-general-response.md). The version w/o nesting was also discussed in the ADR ([here](https://github.com/mozilla-services/merino-py/blob/main/docs/adr/0002-merino-general-response.md#b5-custom-details-field-without-the-provider-nesting)), feel free to take a read to see if that makes sense to you :)